packages/mobile/src/screens/edit-playlist-screen/EditPlaylistForm.tsx

```diff
@@ -0,0 +1,215 @@
+  const styles = useStyles()
+  const navigation = useNavigation()
+  const dispatch = useDispatch()
+
+  const openDeleteDrawer = useCallback(() => {
+    dispatch(openDeletePlaylist({ playlistId }))
+    navigation.goBack()
+  }, [dispatch, playlistId, navigation])
+
+  const trackIds = useMemo(
+    () => values.tracks?.map(({ track_id }) => track_id),
+    [values.tracks]
+  )
+
+  const handleReorder = useCallback(
+    ({ data, from, to }) => {
+      const reorder = [...values.track_ids]
+      const tmp = reorder[from]
+      reorder.splice(from, 1)
+      reorder.splice(to, 0, tmp)
+
+      const reorderedTracks = data.map((id: number) =>
+        values.tracks?.find((t) => t.track_id === id)
+      )
+
+      setFieldValue('track_ids', reorder)
+      setFieldValue('tracks', reorderedTracks)
+    },
+    [setFieldValue, values.track_ids, values.tracks]
+  )
+
+  const handleRemove = useCallback(
+    (index: number) => {
+      if ((values.track_ids.length ?? 0) <= index) {
+        return
+      }
+      const {
+        track: trackId,
+        metadata_time: metadataTime,
+        time
+      } = values.track_ids[index]
+
+      const trackMetadata = values.tracks?.find(
+        ({ track_id }) => track_id === trackId
+      )
+
+      if (!trackMetadata) return
+
+      setFieldValue('removedTracks', [
+        ...values.removedTracks,
+        { trackId, timestamp: metadataTime ?? time }
+      ])
+
+      const tracks = [...(values.tracks ?? [])]
+      tracks.splice(index, 1)
+
+      setFieldValue('tracks', tracks)
+    },
+    [values.track_ids, values.tracks, values.removedTracks, setFieldValue]
+  )
+
+  const header = (
+    <View style={styles.header}>
+      <PlaylistImageInput />
+      <PlaylistNameInput />
+      <PlaylistDescriptionInput />
+    </View>
+  )
+
+  const footer = (
+    <View style={styles.footer}>
+      <TextButton
+        variant='neutralLight4'
+        title={messages.deletePlaylist}
+        icon={IconTrash}
+        iconPosition='left'
+        onPress={openDeleteDrawer}
+        styles={{
+          root: styles.deleteButtonRoot,
+          text: styles.deleteButtonText
+        }}
+      />
+    </View>
+  )
+
+  return (
+    <FormScreen
+      onSubmit={handleSubmit}
+      onReset={handleReset}
+      goBackOnSubmit
+      title={messages.screenTitle}
+      topbarLeft={
+        <TopBarIconButton
+          icon={IconCaretLeft}
+          style={styles.backButton}
+          onPress={navigation.goBack}
+        />
+      }
+    >
+      <KeyboardAwareScrollView>
+        <Tile style={styles.tile}>
+          {header}
+          <Divider />
+          {values.tracks?.length ? (
+            <>
+              <View>
+                {values.tracks.length ? (
+                  <Text style={styles.reorderTitle}>
+                    {messages.reorderTitle}
+                  </Text>
+                ) : null}
+              </View>
+              <TrackList
```

Comment:
is it possible to move the handleReorder/handleRemove logic to a "DraggableTrackList" component that can manage itself? that way this form can just focus on the core layout? small architecture thing, lmk what you think.

Reply:
moved this code over from the previous implementation. it seems like the reordering has to update form values so it seems like that's why the callback was defined here, but down to discuss further if you have ideas about this